omegatrade-app/backend/app/models/simulation.model.js

```diff
@@ -0,0 +1,83 @@
+'user strict';
+const database = require('./../config/database.js');
+const { v4: uuidv4 } = require('uuid');
+const Simulation = function () { };
+
+Simulation.getAll = async function (cb) {
+    try {
+        const [result] = await database.run({ sql: 'select sml.sId as sId,sml.companyId as companyId, sml.status as status, cy.companyName as companyName, cy.companyShortCode as companyShortCode from simulations sml LEFT JOIN companies cy ON sml.companyId = cy.companyId LIMIT 3', json: true });
+        cb(null, result)
+    } catch (error) {
+        cb(error, null)
+    }
+}
+
+Simulation.findById = async function (param, cb) {
+    try {
+        const sId = param.sId
+        const [ result ] = await database.run({
+            sql: 'select sId,companyId,companyName,companyShortCode,status from simulations where sId = @sId',
+            params: {
+                sId: sId
+            },
+            json: true
+        });
+        cb(null, result)
+        
+    } catch (error) {
+        cb(error, null)
+    }
+}
+
+Simulation.findByCompanyId = async function (companyId, sid) {
+    try {
+        const [result] = await database.run({
+            sql: 'select sId,companyId,companyName,companyShortCode,status from simulations where companyId = @companyId and sid = @sid',
+            params: {
+                companyId: companyId,
+                sid: sid
+            },
+            json:true
+        });
+       return result;
```

Comment:
Here we are finding by companyId and sid both and then also function name used is findByCompanyId?

Reply:
yes, Agreed Simulation.findOne is the correct function Name.